compiler/damlc/tests/src/DA/Test/Packaging.hs

```diff
@@ -729,8 +731,13 @@ tests Tools{damlc} = testGroup "Packaging" $
           exitCode @?= ExitFailure 1
           assertBool ("Expected \"non-exhaustive\" error in stderr but got: " <> show stderr) ("non-exhaustive" `isInfixOf` stderr)
 
-    , testCaseSteps "data-dependencies + exposed-modules" $ \step -> withTempDir $ \projDir -> do
-          step "Building dependency"
+    , testProperty "data-dependencies + exposed-modules" $ \(n1 :: Int) (n2 :: Int) ->
+        -- Since the order in which dependencies are processed depends on their PackageIds,
```

Comment:
This is necessary since the test passed even before adding the edge `data-dependency -> dependency` because the `PackageId` calculated for `dependency` happened to be lower than the one for `data-dependency`. 

I couldn't find a way to combine `testProperty` with `testCaseSteps`, so we lose the `step` functionality, but maybe that isn't so bad for one test.

Reply:
Nice test! I'm a little bit worried about using a property test though because of how long it takes to build with data-dependencies. But I guess we'll see if we start running into test timeouts again.